pkg/arvo/app/chat-store.hoon

```diff
@@ -206,15 +213,10 @@
   ?~  envelopes.act
     :_  state(inbox (~(put by inbox) path.act u.mailbox))
     %+  send-diff  path.act
-    :*  %messages
-        path.act
-        (sub length.config.u.mailbox (lent evaluated-envelopes))
-        length.config.u.mailbox
-        evaluated-envelopes
-    ==
+    [%messages path.act 0 (lent evaluated-envelopes) evaluated-envelopes]
```

Comment:
I think we still want to pass the old-style indexes? ie the ones that match message numbers?
Your range here is always going to be fixed to start at `0`, which doesn't really tell you anything about the messages.

Arguably we don't want to send them at all, because they're already in the message's numbers. I guess this is somewhat similar to the argument @belisarius222 makes in chat-hook.

(We don't have any ask-for-messages-by-index watch/scry paths, do we? If we did, we'd want those to use message numbers, not list indexes.)

Reply:
This still stands I think. With these changes, it looks like `+peek-x-envelopes` expects you to know about the reverse ordering in storage, which 1) leaks an implementation detail and 2) isn't practically usable as implemented right now, because to get at _specific_ messages, you need to know about the total that exists, and adjust the indexes you're asking about accordingly. (Though maybe the negative number addressing is still usable?)

If there ever was a time to document these interfaces, even just one-liners in app file headers, this is probably it.

Reply:
It works. If we want to do refactors for developer ergonomics, let's do them separately.